### PR TITLE
Align receipt with execution apis

### LINKED ISSUE #302: `eth_getTransactionReceipt` invalid response: hex string instead of uint
### Description

When performing `eth_getTransactionReceipt` the response is not valid. According to https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.yaml#L50-L101 the `blockNumber`, `cumulativeGasUsed`, `gasUsed`, `transactionIndex`, `effectiveGasPrice` and `status` properties are `uint`, the relay response uses hex strings instead. 

A json-rpc client with client-side validation, such as go-ethereum, throws validation exceptions in such cases.

### Steps to reproduce

```sh
curl -X POST $BASE_URL \
                                       -H 'Content-Type: application/json' \
                                       -d "{\"id\":\"2\",\"jsonrpc\":\"2.0\",\"method\":\"eth_getTransactionReceipt\",\"params\":[\"$TRANSACTION_HASH\"]}" --silent
```
`BASE_URL=http://localhost:7546`
`TRANSACTION_HASH=0x9040d4244cfeb7c5fc96ac378308d9df8a6616c550e9d02162b3ef3fe889df86`

### Additional context

image

image

### Hedera network

testnet

### Version

main

### Operating system

Linux

### PR DESCRIPTION
**Pull request by shemnon**
**Description**:
Move uints back to hex strings

Signed-off-by: Danno Ferrin <danno.ferrin@hedera.com>

**Related issue(s)**:

Fixes #302

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/model.ts [hunk 1]
```diff
@@ -103,36 +104,36 @@ export class CachedBlock extends Block {
...
         // TODO this.to = record.contractFunctionResult?.contractId;
-        this.cumulativeGasUsed = Number(gasUsed);
-        this.gasUsed = Number(gasUsed);
+        this.cumulativeGasUsed = gasUsed;
+        this.gasUsed = gasUsed;
         this.contractAddress = contractAddress;
         this.logs = [];
         this.logsBloom = '';
-        this.status = record.receipt.status == Status.Success ? 1 : 0;
+        this.status = record.receipt.status == Status.Success ? "0x1" : "0x0";
```
Comment (Nana-EC) [inline packages/relay/src/lib/model.ts:136]: Also move `0x1` to a static class variable
```suggestion
        this.status = record.receipt.status == Status.Success ? "0x1" : EthImpl.zeroHex;
```

File: packages/relay/src/lib/model.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 // Used for fake implementation of block history
 import {Status, TransactionRecord} from "@hashgraph/sdk";
+import {EthImpl} from "./eth";
```
Comment (Nana-EC) [inline packages/relay/src/lib/model.ts:23]: Hmm, seems weird to be importing this just for a formatting method.
Eventually (not a fix for now) I think we should move the converter functions out to the `formatter.ts` class

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Looking good.
Some suggestions also i think you're missing test update.
Can see [my PR](https://github.com/hashgraph/hedera-json-rpc-relay/pull/315/files#diff-febf1674c0e8b4be0c0f626009660fb57c4160a97d7be3f97b80e38a93aa5016) that introduced the change so it's easy to see what to revert

File: packages/relay/src/lib/model.ts [hunk 3]
```diff
@@ -103,36 +104,36 @@ export class CachedBlock extends Block {
...
 
     constructor(txHash:string, record:TransactionRecord, block:Block) {
-        const gasUsed = record.contractFunctionResult == null ? 0 : record.contractFunctionResult.gasUsed;
+        const gasUsed = EthImpl.numberTo0x(record.contractFunctionResult == null ? 0 : record.contractFunctionResult.gasUsed.toNumber());
         const contractAddress = record.contractFunctionResult == undefined ? undefined : "0x" + record.contractFunctionResult.contractId?.toSolidityAddress();
 
         this.transactionHash = txHash;
-        this.transactionIndex = 0;
-        this.blockNumber = Number(block.number);
+        this.transactionIndex = "0x0";
```
[the remark this answers is not in the document]
Comment (shemnon) [reply packages/relay/src/lib/model.ts]: I opted for a direct rollback on this line.

File: packages/relay/src/lib/model.ts (hunk 1 above)
Comment (shemnon) [reply packages/relay/src/lib/model.ts:135]: I opted for a direct rollback on this line.

Comment (Nana-EC) [reply packages/relay/src/lib/model.ts:135]: Gotcha. `0x0` and `0x1` should still be static class constants

Comment (Nana-EC) [reply packages/relay/src/lib/model.ts:135]: NO biggie. We can clean this and other files up later

**Review by Nana-EC — COMMENTED** [review]
LG.
"should execute "eth_getTransactionReceipt" for hash..."  acceptance tests need updating with your new logic

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**